datafusion/physical-plan/src/topk/mod.rs

```diff
@@ -232,6 +248,105 @@ impl TopK {
         // which means the top K won't change and the computation can be finished early.
         self.attempt_early_completion(&batch)?;
 
+        if updated {
+            // update the filter representation of our TopK heap
+            self.update_filter()?;
+        }
+
+        Ok(())
+    }
+
+    /// Update the filter representation of our TopK heap
+    fn update_filter(&mut self) -> Result<()> {
+        let Some(filter) = &self.filter else {
+            return Ok(());
+        };
+        if let Some(thresholds) = self.heap.get_threshold_values(&self.expr)? {
```

Comment:
(minor) You could probably remove another level of indent here with the same pattern.

```suggestion
        let Some(thresholds) = self.heap.get_threshold_values(&self.expr)? else {
          return Ok(());
         }
```